src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/TransactionValidator.cs

```diff
@@ -0,0 +1,54 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Health.Fhir.Core.Exceptions;
+using Microsoft.Health.Fhir.Core.Models;
+using static Hl7.Fhir.Model.Bundle;
+
+namespace Microsoft.Health.Fhir.Api.Features.Resources.Bundle
+{
+    public static class TransactionValidator
+    {
+        public static void ValidateTransaction(HashSet<string> resourceIdList, EntryComponent entry)
+        {
+            if (ValidateBundleEntry(entry))
+            {
+                string resourceId = GetResourceUrl(entry);
+
+                if (resourceId != null)
```

Comment:
What would the behavior be of a `FullUrl` of `""` on a `POST` be?

Reply:
It should skip that case as well. I have now updated the check.